client/web/src/components/FilteredConnection/ConnectionNodes.tsx

```diff
@@ -93,111 +93,108 @@ interface ConnectionNodesProps<C extends Connection<N>, N, NP = {}, HP = {}>
-                                    </span>
-                                ) : (
-                                    'total'
-                                )}
-                            </span>{' '}
-                            {this.props.connection.nodes.length < totalCount &&
-                                `(showing first ${this.props.connection.nodes.length})`}
-                        </small>
-                    </p>
-                )
-            } else if (this.props.connection.pageInfo?.hasNextPage) {
-                // No total count to show, but it will show a 'Show more' button.
-            } else if (totalCount === 0) {
-                summary = this.props.emptyElement || (
-                    <p className="filtered-connection__summary" data-testid="summary">
-                        <small>
-                            No {this.props.pluralNoun}{' '}
-                            {this.props.connectionQuery && (
-                                <span>
-                                    matching <strong>{this.props.connectionQuery}</strong>
-                                </span>
-                            )}
-                        </small>
-                    </p>
-                )
-            }
-        }
-
-        const nodes = this.props.connection.nodes.map((node, index) => (
-            <NodeComponent key={hasID(node) ? node.id : index} node={node} {...this.props.nodeComponentProps!} />
-        ))
-
-        return (
-            <>
-                {this.props.connectionQuery && summary}
-                {this.props.connection.nodes.length > 0 && (
-                    <ListComponent
-                        className={classNames('filtered-connection__nodes', this.props.listClassName)}
-                        data-testid="nodes"
-                    >
-                        {HeadComponent && (
-                            <HeadComponent
-                                nodes={this.props.connection.nodes}
-                                totalCount={this.props.connection.totalCount}
-                                {...this.props.headComponentProps!}
-                            />
-                        )}
-                        {ListComponent === 'table' ? <tbody>{nodes}</tbody> : nodes}
-                        {FootComponent && <FootComponent nodes={this.props.connection.nodes} />}
-                    </ListComponent>
-                )}
-                {!this.props.connectionQuery && summary}
-                {!this.props.loading && !this.props.noShowMore && hasNextPage && (
-                    <ConnectionNodesSummaryShowMore
-                        onShowMore={this.props.onShowMore}
-                        showMoreClassName={this.props.showMoreClassName}
-                    />
-                )}
-            </>
-        )
+export const getTotalCount = <N,>(connection: Connection<N>, first: number): number | null => {
```

Comment:
Optional suggestion that might make some of this code more readable:

```suggestion
export const getTotalCount = <N,>({ totalCount, nodes, pageInfo }: Connection<N>, first: number): number | null => {
```